be/src/olap/tablet_manager.cpp

```diff
@@ -976,13 +978,17 @@ OLAPStatus TabletManager::build_all_report_tablets_info(std::map<TTabletId, TTab
                     expire_txn_map.erase(find);
                 }
                 t_tablet.tablet_infos.push_back(tablet_info);
+                if (tablet_ptr->tablet_id() == tablet_id) {
+                    DorisMetrics::instance()->tablet_version_num_distribution->add(tablet_ptr->version_count());
```

Comment:
It's not atomic and may last a long time before all values added, that means we may get a middle state which is not very correct.

Reply:
> It's not atomic and may last a long time before all values added, that means we may get a middle state which is not very correct.

Thank you, you are right.